src/monodroid/jni/embedded-assemblies.c

```diff
@@ -319,6 +320,13 @@ register_debug_symbols_for_assembly (struct DylibMono *mono, const char *entry_n
 	return 1;
 }
 
+MONO_API int
+monodroid_embedded_assemblies_set_assemblies_prefix (const char *prefix)
+{
+	assemblies_prefix = prefix;
```

Comment:
Right idea. Not quite right implementation (possibly).

As a *caller*, what are the lifetime semantics of `prefix`? Answer: Don't free it; if you do, things will break.

Is that the *correct* answer? Generally not, *especially* with a `const char*` argument type.

The saner thing to do, then, is to **strdup**(3) the value, ensuring that if the caller free's `prefix`, `assemblies_prefix` is still valid. This raises another problem: to avoid a memory leak, `monodroid_embedded_assemblies_set_assemblies_prefix()` needs to **free**(3) the existing value, which in turn means it can't be given a string constant, as is currently done.

Thus:

1. Use **strdup**(3) and **free**(3) in `monodroid_embedded_assemblies_set_assemblies_prefix()`
2. Add a new `get_assemblies_prefix()` function.
3. Use `get_assemblies_prefix()` in `gather_bundled_assemblies_from_apk()`/etc., *not* `assemblies_prefix`.

Thus:

```C
static  char                         *assemblies_prefix;

MONO_API int
monodroid_embedded_assemblies_set_assemblies_prefix (const char *prefix)
{
	free (assemblies_prefix);
	assemblies_prefix = strdup (prefix);
}

static const char *
get_assemblies_prefix (void)
{
	if (!assemblies_prefix) {
		assemblies_prefix = strdup ("assemblies/");
	}
	return assemblies_prefix;
}
```